pkg/services/ngalert/state/historian/loki.go

```diff
@@ -2,6 +2,7 @@ package historian
 
 import (
 	"context"
+	"crypto/md5"
```

Comment:
If the algorithm is mostly irrelevant, can we use one of the SHA families instead? Those are supported in the browser's [webcrypto API ](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#parameters) and might allow us to use the same fingerprinting on the front-end for areas where we don't have access to data from the Alert State History API.

Reply:
Check @JohnnyQQQQ's comment: https://github.com/grafana/grafana/pull/65968#discussion_r1162972006

I'm thinking of instead just directly lifting prometheus's hash function. It's just a basic FNV hash and we can just lift its exact same base and offset. This gives us some stronger alignment with prometheus, and the UI can also easily implement this and get the same values - WDYT? There's no real reason to use any crypto hash here.

Reply:
I ended up going this route. The hash function we used is implemented here: https://github.com/prometheus/common/blob/main/model/fnv.go
And it serializes the label set using: https://github.com/prometheus/common/blob/2f04d2ec94b96df44aa4f8a60ae9fb29d8512636/model/signature.go#L33

This should translate pretty easily, it's faster, and more universal.